src/grass_gis_helpers/general.py

```diff
@@ -111,3 +111,20 @@ def free_ram(unit, percent=100):
         return int(round(memory_gb_percent))
     else:
         grass.fatal(f"Memory unit {unit} not supported")
+
+
+def test_memory(memory_string):
+    """Function returns available memory.
```

Comment:
```suggestion
    """Test if desired memory is available. In case RAM is smaller than desired memory, use free RAM instead of desired memory value.
```